### PR TITLE
Simplify Min/Max using ShapeEnv value ranges

### LINKED ISSUE #137096: Expressions involving Min/Max don't simplify with value ranges information
### 🐛 Describe the bug

Repro:

```
import torch

torch._dynamo.config.capture_scalar_outputs = True

@torch.compile(fullgraph=True)
def f(x):
    x0, x1 = x.tolist()
    torch._check(x0 <= 5)
    if torch.sym_min(x0, 5) == x0:
        return torch.tensor(True)
    else:
        return torch.tensor(False)

f(torch.tensor([3, 5]))
```

Here, because we know that x0 <= 5, we can simplify Min(x0, 5) to just x0. However, it is a little awkward to actually implement this, because today Sympy expressions don't necessarily have access to the ShapeEnv, so we can't actually put the simplification there. We could add a dedicated simplification algorithm in ShapeEnv.simplify to do this, I guess. I kind of wish I did have access to these ranges inside the Sympy simplification rules though.

This is similar to https://github.com/pytorch/pytorch/issues/125914 but it is in some sense simpler, as no size oblivious is involved (so it would be valid to unconditionally do the simplification here; in the other case, we must not simplify ahead of time and only apply the simplification when a size oblivious test is done.)

### Versions

main

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* __->__ #186248

ShapeEnv.simplify previously only had narrow handling for Max(0, x) and
Max(1, x), so value-range facts such as torch._check(u0 <= 5) were not used to
simplify Min(u0, 5). That left guards like Eq(Min(5, u0), u0) data-dependent in
Dynamo even though the ordering was already known.

Generalize the ShapeEnv simplification pass to visit both Min and Max atoms and
remove dominated arguments when ShapeEnv can prove their ordering from value
ranges. The current implementation keeps the reasoning in ShapeEnv, where range
information is available, and uses cheap lower-bound/sign checks before the
heavier bound_sympy path so common Min/Max simplifications do not pay the full
static solver cost. The legacy _maybe_evaluate_static fallback remains limited to
preserving the previous Max(0/1, x) behavior.

This pass runs only in the non-size-oblivious path, preserving the existing
size-oblivious guard behavior. A narrower binary-only change would fix the exact
repro, but using the same proved-ordering rule for n-ary Min/Max avoids leaving
equivalent flattened expressions unsimplified while still requiring ShapeEnv to
prove every removed argument.

Fixes #137096

Generated by my agent

Benchmark Results:

- Mixed cold simplify workload with 400 legacy Max expressions and 400 newly
  handled Min expressions: the earlier solver-heavy PR version was 0.340158s
  best of 5; after the cheap ordering checks and fallback tightening this is
  0.069140s best of 5.
- Unproven Min/Max stress workload with 500 expressions: 0.243360s best of 5
  after fallback tightening, compared with 0.537786s best of 3 before the
  narrowing in that follow-up.
- Legacy Max-only cold simplify workload stayed roughly unchanged in earlier
  benchmarking: main best of 5 was 0.122475s and the patched path was 0.125108s.

Test Plan:

- Reproduced the issue repro before the fix; it failed with
  `Could not guard on data-dependent expression Eq(Min(5, u0), u0)`.
- Re-ran the issue repro after the fix; it printed `tensor(True)`.
- `python test/test_dynamic_shapes.py TestPySymInt.test_min_max_simplify_with_value_ranges TestDimConstraints.test_simplify_max_1_0`
- `python test/dynamo/test_misc.py MiscTests.test_sym_min_simplifies_with_checked_upper_bound`
- `python test/inductor/test_indexing.py -k test_simplify_with_ranges_forwards_local_ranges_to_shape_env`
- `timeout 600 python test/export/test_db.py -v`
- `git diff --check`
- `lintrunner -a`

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @azahed98

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I updated the Dynamo regression test to bind the intentionally unused second scalar as `_` in test/dynamo/test_misc.py, then rebased and resubmitted the PR.

**Comment by jansel** [comment]
My agent says the failing inductor HuggingFace CI shard was caused by DebertaV2ForMaskedLM failing during eager Transformers import before the existing accuracy skip could apply. I updated benchmarks/dynamo/huggingface.yaml to skip that model before load, then rebased and resubmitted the PR.

**Review by aorenste — APPROVED** [review]

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by jansel** [comment]
My agent says I updated the PR after the repeated Windows trunk merge failure in win-vs2022-cpu-py3 test_decomp. The change keeps the Min/Max value-range simplification but adds cheap lower-bound/range checks before the expensive static solver, improving the synthetic mixed simplify workload from 0.340158s to 0.124761s and preserving the local-range regression test.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I updated the PR for the current CI failures. The Inductor Triton CPU failure was an inline expected-output change from the intended Min/Max simplification, so I refreshed that golden. I also narrowed the Min/Max proof path to avoid constructing/solving unproven comparisons except for the legacy Max(0/1, x) fallback; locally this keeps the issue repro passing, passes export/test_db, and improves the mixed simplify benchmark to 0.069140s best of 5.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I updated the PR for the current CI failures. The related CPU/CUDA/ROCm strided-block failures were from a stale inline expected-output update, so I restored that test to the main-branch 3D broadcast expectation that CI actually generated. I also documented why the cheap Min/Max ordering checks intentionally run before bound_sympy; fresh CI is now pending.

**Comment by jansel** [comment]
My agent says I refreshed the PR description to match the current optimized Min/Max proof path, latest benchmark numbers, and validation; no code changed.
